frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/wizard-pages/storage-and-nodes-step.tsx

```diff
@@ -37,31 +48,135 @@ import { State, Action } from '../state';
 import AttachedDevicesNodeTable from '../../sc-node-list';
 import { PVsAvailableCapacity } from '../../../pvs-available-capacity';
 import { getSCAvailablePVs } from '../../../../../selectors';
-import { pvResource } from '../../../../../constants/resources';
+import { nodeResource, pvResource, scResource } from '../../../../../constants/resources';
 
-const validate = (scName: string, enableMinimal: boolean, nodes: NodeKind[]): Validation[] => {
+const validate = (
+  scName: string,
+  enableMinimal: boolean,
+  nodes: NodeKind[],
+  stretchClusterChecked: boolean,
+): Validation[] => {
   const validations = [];
   if (enableMinimal) {
     validations.push(VALIDATIONS.MINIMAL);
   }
   if (!scName) {
     validations.push(VALIDATIONS.BAREMETALSTORAGECLASS);
   }
-  if (scName && nodes.length < MINIMUM_NODES) {
+  if (scName && !stretchClusterChecked && nodes.length < MINIMUM_NODES) {
     validations.push(VALIDATIONS.MINIMUMNODES);
   }
   return validations;
 };
 
+export const StretchClusterFormGroup: React.FC<stretchClusterFormGroupProps> = ({
+  state,
+  dispatch,
+  pvData,
+  nodesData,
+}) => {
+  const { stretchClusterChecked, selectedArbiterZone, storageClass, nodes } = state;
+  const [zonesOption, setZonesOptions] = React.useState({});
+  const [zones, setZones] = React.useState([]);
+  const [showInfoAlert, setShowInfoAlert] = React.useState(true);
+  const [storageClassData, storageClassLoaded, storageClassError] = useK8sWatchResource<
+    StorageClassResourceKind[]
+  >(scResource);
+
+  const isArbiterDisabled = (): boolean => {
+    const isDisabled: boolean = true;
+    if (!storageClassError && storageClassData.length && storageClassLoaded) {
+      return !storageClassData.some((sc) => isArbiterSC(sc, pvData, nodesData));
+    }
+    return isDisabled;
+  };
+
```

Comment:
```suggestion
const nodesDataMemoized = useDeepCompareMemoize(nodesData, true);
```
Use this inside useEffect below.